modules/http/src/main/java/org/mule/runtime/module/http/internal/listener/DefaultHttpListener.java

```diff
@@ -154,13 +167,46 @@ public void responseSendSuccessfully() {}
   }
 
   private Event createEvent(HttpRequestContext requestContext) throws HttpRequestParsingException {
-    Event muleEvent =
-        HttpRequestToMuleEvent.transform(requestContext, muleContext, flowConstruct, parseRequest, listenerPath);
+    Event muleEvent = Event.builder(create(flowConstruct, resolveUri(requestContext).toString())).message(HttpRequestToMuleEvent
+        .transform(requestContext, SystemUtils.getDefaultEncoding(muleContext), parseRequest, listenerPath))
+        .exchangePattern(REQUEST_RESPONSE).flow(flowConstruct).session(new DefaultMuleSession()).build();
```

Comment:
This is just moved code from an existent method. We still need that in the builder for transport compatibility. This specific usage will be removed once we get rid of the http module.